app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
-                }
-            }
+            this.templateService.getTemplates(this.applicationId, this.actionId).subscribe((templates) => {
+                this.jobTemplate = templates.job;
+                this.poolTemplate = templates.pool;
+                this.title = `Run ${this.actionId} from ${this.applicationId}`;
+                this.createForms();
+            });
         });
     }
-
-    public createForms() {
-        let parameterKeys = Object.keys(this.jobTemplate.parameters);
+    private createForms() {
+        let jobParameters = [];
+        let poolParameters = [];
+        if (this.jobTemplate && this.jobTemplate.parameters){
+            jobParameters = Object.keys(this.jobTemplate.parameters);
+        }
+        if (this.poolTemplate && this.poolTemplate.parameters){
+            poolParameters = Object.keys(this.poolTemplate.parameters);
+        }
         let fg = {};
-        for (let key of parameterKeys) {
+
+        for (let key of jobParameters) {
             if ("defaultValue" in this.jobTemplate.parameters[key]) {
                 const defaultValue = String(this.jobTemplate.parameters[key].defaultValue);
                 fg[key] = new FormControl(defaultValue);
             } else {
                 fg[key] = new FormControl();
             }
         }
-        this.jobFormGroup = this.formBuilder.group(fg);
-
-        this.form = this.formBuilder.group({
-            job: this.jobFormGroup,
-            pool: this.pickedPool,
-        });
-        if (this._formValue) {
-            this.form.setValue(this._formValue);
+        for (let key of poolParameters) {
+            if ("defaultValue" in this.poolTemplate.parameters[key]) {
+                const defaultValue = String(this.poolTemplate.parameters[key].defaultValue);
+                fg[key] = new FormControl(defaultValue);
+            } else {
+                fg[key] = new FormControl();
+            }
         }
-        this.form.valueChanges.debounceTime(400).distinctUntilChanged().subscribe((newFormValue) => {
-            this.router.navigate([], {
-                relativeTo: this.route,
-                queryParams: {
-                    formParams: JSON.stringify(newFormValue),
-                },
-            });
-        });
-    }
-
-    public getContainerFromFileGroup(fileGroup: string) {
-        return fileGroup && `fgrp-${fileGroup}`;
+        this.form = new FormGroup(fg);
     }
 
-    @autobind()
-    public submit() {
-        this.error = null;
-        const jobParams = this.jobFormGroup.value;
-        // RPC takes in Template JSON object and Parameter JSON object
-        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this._buildJobTemplate(), jobParams]);
+    private runJobWithPool(expandedPoolTemplate, jobInputs){
+        delete expandedPoolTemplate.id;
+        this.jobTemplate.job.properties.poolInfo = {
+            autoPoolSpecification: {
+            autoPoolIdPrefix: "jobname",
```

Comment:
should the prefix be the actual job name, not the string "jobname"?

Also, consider what kinds of constraints you will run into here. Batch has limitations on what values are allowed for this. It would be worth taking a look at the Batch docs to know what issues you may run into: https://docs.microsoft.com/en-us/rest/api/batchservice/